tests/test_admin.py

```diff
@@ -248,6 +261,64 @@ def test_advanced_settings_link(self):
         )
 
 
+class SetHomeViewTestCase(CMSTestCase):
+    def test_get_method_is_not_allowed(self):
+        pagecontent = PageContentWithVersionFactory()
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.get(
+                self.get_admin_url(PageContent, "set_home_content", pagecontent.pk)
+            )
+        self.assertEqual(response.status_code, 405)
+
+    def test_root_page_is_allowed_to_set_home(self):
+        version = PageVersionFactory(content__page__node__depth=1)
+        pagecontent = version.content
+        version.publish(self.get_superuser())
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.post(
+                self.get_admin_url(PageContent, "set_home_content", pagecontent.pk)
+            )
+        url = reverse("admin:cms_pagecontent_changelist")
+
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, url)
+
+    def test_non_root_page_should_not_allowed_to_set_home(self):
+        version = PageVersionFactory(content__page__node__depth=2)
+        pagecontent = version.content
+        version.publish(self.get_superuser())
```

Comment:
any reason you don't just define `version = PageVersionFactory(state=PUBLISHED, ...)`?